jwql/website/apps/jwql/static/js/jwql.js

```diff
@@ -150,4 +157,93 @@ function sort_by_thumbnails(sort_type) {
     } else if (sort_type == 'Exposure Start Time') {
         tinysort(thumbs, {attr:'exp_start'});
     }
+};
+
+/**
+ * Updates the thumbnail-filter div with filter options
+ * @param {Object} data - The data returned by the update_thumbnails_page AJAX method
+ */
+function update_filter_options(data) {
+
+    for (var i = 0; i < Object.keys(data.dropdown_menus).length; i++) {
+
+        // Parse out useful variables
+        filter_type = Object.keys(data.dropdown_menus)[i];
+        filter_options = Array.from(new Set(data.dropdown_menus[filter_type]));
+        num_rootnames = Object.keys(data.file_data).length;
+        dropdown_key_list = Object.keys(data.dropdown_menus);
+
+        // Build div content
+        content = '<div class="mr-4">';
+        content += 'Show only ' + filter_type + ':';
+        content += '<div class="dropdown">';
+        content += '<button class="btn btn-primary dropdown-toggle" type="button" id="' + filter_type + '_dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"> All ' + filter_type + 's </button>';
+        content += '<div class="dropdown-menu" aria-labelledby="dropdownMenuButton">';
+        content += '<a class="dropdown-item" href="#" onclick="show_only(\'' + filter_type + '\', \'All ' + filter_type + 's\', \'' + dropdown_key_list + '\',\'' + num_rootnames + '\');">All ' + filter_type + 's</a>';
+
+        for (var j = 0; j < filter_options.length; j++) {
+            content += '<a class="dropdown-item" href="#" onclick="show_only(\'' + filter_type + '\', \'' + filter_options[j] + '\', \'' + dropdown_key_list + '\', \'' + num_rootnames + '\');">' + filter_options[j] + '</a>';
+        };
+
+        content += '</div></div>';
+    };
+
+    // Add the content to the div
+    $("#thumbnail-filter")[0].innerHTML = content;
+};
+
+/**
+ * Updates the img_show_count component
+ * @param {Object} data - The data returned by the update_thumbnails_page AJAX method
+ */
+function update_show_count(data) {
+
+    num_rootnames = Object.keys(data.file_data).length;
+    content = 'Showing ' + num_rootnames + '/' + num_rootnames + ' activities';
+    content += '<a href="https://jwst-docs.stsci.edu/display/JDAT/File+Naming+Conventions+and+Data+Products" target="_blank" style="color: black">';
+    content += '<span class="help-tip mx-2">i</span></a>';
+    $("#img_show_count")[0].innerHTML = content;
```

Comment:
I think I'm having a little trouble working out why, for the `thumbnails.html` page, divs for the `img_show_count` and sort options are written here in `jwql.js`, but similar/identical divs are still written in the HTML for `archive.html`. I would expect they would be handled in the same place? Sorry if I'm missing something simple!

Reply:
@laurenmarietta Good point.  Probably this javascript function could be generalized and used in both templates.  I'll try to do that.